src/core/database/property-updater.service.ts

```diff
@@ -65,7 +65,7 @@ export class PropertyUpdaterService {
     session: ISession;
     object: TObject;
     key: Key;
-    value: UnwrapSecured<TObject[Key]>;
+    value?: UnwrapSecured<TObject[Key]>;
```

Comment:
I don't think this is right. If there's no value then you shouldn't be calling `updateProperty`. Thoughts?

Reply:
Going to look into this one after addressing https://github.com/SeedCompany/cord-api-v3/pull/149#discussion_r384056579

Reply:
Ah okay so I'm running into an odd issue where doing an `== undefined` assertion doesn't seem to work, most likely because we're doing an `await` call on `this.updateProperty` so TS is assuming that since `await` runs on an asynchronous thread, the value of `changes[prop]` could have changed by the time the `await` is executed. This is my faulty code as of right now:

```
      if (
        changes[prop] == null || changes[prop] == undefined ||
        unwrapSecured(object[prop]) === changes[prop]
      ) {
        continue;
      }
```

but I'm still getting an error on the `value: changes[prop]` line:
```
Type 'UnwrapSecured<TObject[keyof TObject]> | undefined' is not assignable to type 'UnwrapSecured<TObject[keyof TObject]>'.
  Type 'undefined' is not assignable to type 'UnwrapSecured<TObject[keyof TObject]>'
```

Reply:
FYI `foo == null` is the same as `foo === null || foo === undefined`. So the second _or_ condition is redundant. (I know you were probably just trying stuff).

Ok so it's because `Secured['value']` is `T | undefined`. To fix you can wrap in `NonNullable` to strip the `undefined`.
```ts
NonNullable<UnwrapSecured<TObject[keyof TObject]>>
```

BUTTTT thinking through it more, I'm not so sure we should be requiring a value. If some properties are nullable then I think they should be able to be "reset" to `null` explicitly. Lemme talk to Michael and get his thoughts.

Reply:
I completely blanked on the double equal operator, so used to just using `===` without thought LOL.

And sounds good! As of right now, there are no type errors and the tests are passing. I think this is the only outstanding issue, lemme know how to move forward after you talk to Michael.

Reply:
Ok after talking on slack let's just leave this for now. We can validate not null is what we want in API validation